### ISSUE TITLE
Word Embedding tune

### ISSUE THREAD

**Issue by MariamEssam** [taoshen58/DiSAN#14]
Have you tried to use other word embedding either then Glove?I have tried to replace the current word embedding with other one but I got surprised with the performance that's been dropped to become round 47% 

1. I make all the token in SST unique each one has row and column after with sentence it lies and generated for each of them a new word embedding
2. I made sure that 'token_seq_digital' has the right indices(I am pretty sure it points on the right one)
3. The dictionary used by the train, test and dev contains all the token as they become unique 
4. emb_mat_token is now for all the new emb vector the equivalent vector for the dictionary token 
The new word_embedding_length now is 900. Do you have a clue why the performance is dropped in this manner? I expected some enhancement or even the same results but not less.

**Comment by taoshen58** [comment]
Sorry for the late reply.

I didn't test any other embedding method because it's not the focus of this paper.

Did you try to shorten the embedding dim to reduce the trainable parameters?

**Comment by MariamEssam** [comment]
Thank you for your answer.
1. What do you mean by trying to shorten the embedding dim?
2. Where you train the embedding I can't discover it from the code. I know that you fine-tune glove but can't get where you do that  
3. finally and you can skip this if you think it's out of your scope but shall we fine-tune all the embedding or glove only is what we need to do 
Between I know this drop was happening due to the dimensionality of the embedding because I have tested the same with Glove but in a way, I have made emb_mat contains all the token not only unique ones and yes the performance has dropped sharply 
Thank you again Tao and I will  appreciate if I get a quick response for these questions

**Comment by taoshen58** [comment]
1. I mean too long embedding dim will lead to too many parameters required to train. It will be difficult to reach coverage.
2. I build a trainable word emb matrix which initialized by glove
3. ... I didnt get you

**Comment by MariamEssam** [comment]
I was asking about where in the code we tune the word representations? And in general, if I used another word embedding is it possible to not need this tuning?

**Comment by taoshen58** [comment]
Got u, I learned some basic knowledge about the transferring/pretrained embeddings like ELMo and CoVe within these days. 

The other embeddings can be fine-tuned in this framework. In current glove setup, I create a tensorflow trainable variable which is initialized by the pretrained glove. The variable will be automatically fine-tuned by backpropagation.  

Therefore, in your setup, just set the variables in word embedding layer or pertained LSTM as "trainable".

**Comment by MariamEssam** [comment]
Thank you!
